dhis-2/dhis-services/dhis-service-node/src/test/java/org/hisp/dhis/node/serializer/CsvNodeSerializerTest.java

```diff
@@ -56,96 +52,81 @@
     void CsvFileIsWrittenWhenOnlySimpleNodesAreProvided()
         throws Exception
     {
-        csvNodeSerializer.serialize( createCollectionWithSimpleNodes(), new FileOutputStream( "output.csv" ) );
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        csvNodeSerializer.serialize( createCollectionWithSimpleNodes(), outputStream );
 
-        List<List<String>> values = readTestFile();
+        String[] resultArray = outputStream.toString().split( "\n" );
 
-        assertEquals( 2, values.size() );
-        assertEquals( 3, values.get( 0 ).size() );
-        assertEquals( 3, values.get( 1 ).size() );
-        assertEquals( "\"First simple node\"", values.get( 0 ).get( 0 ) );
-        assertEquals( "\"Second value simple child\"", values.get( 1 ).get( 1 ) );
-        assertEquals( "\"Third simple node\"", values.get( 0 ).get( 2 ) );
+        assertEquals( 2, resultArray.length );
+        assertEquals( "\"First simple node\",\"Second simple node\",\"Third simple node\"", resultArray[0] );
```

Comment:
Out of curiosity 😋 I have not used it yet myself but I wonder if https://junit.org/junit5/docs/5.8.2/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertLinesMatch(java.util.List,java.util.List) or the variants for streams are a good fit since you are asserting lines in the output stream. Did you try using it?

Reply:
I didn't try it, I guess it would work if I transform the output stream into a list first. 
Regarding the method with the stream signature, it supports monadic streams, so I guess it would need to be transformed into a list as well.